xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/email/NotificationEmailUserPreferenceManager.java

```diff
@@ -38,6 +41,40 @@
     /**
      * @param userId id of a user
      * @return the diff type configured for the given user
+     * @deprecated since 14.10, use {@link #getDiffType(UserReference)} instead
      */
+    @Deprecated(since = "14.10")
     NotificationEmailDiffType getDiffType(String userId);
+
+    /**
+     * @param userReference reference of a user
+     * @return the diff type configured for the given user
+     * @since 14.10
+     */
+    @Unstable
+    default NotificationEmailDiffType getDiffType(UserReference userReference)
+    {
+        return getDiffType();
```

Comment:
Is this really a good default implementation? Wouldn't it be better to throw some exception instead of returning a value that is not for the given user to just throw an exception? The same also applies to the other two new methods.

In particular, when you removed the implementation of this method from the default implementation, you would get an endless loop.

Reply:
The only point of the default implementation is retro-compatibility (in case this interface is implemented by an extension based on an older version).

> In particular, when you removed the implementation of this method from the default implementation, you would get an endless loop.

But doing that would not really make any sense.

Reply:
Okay, I just wasn't sure if it was okay to return possibly incorrect values from the default implementation.